book.rb

```diff
@@ -0,0 +1,18 @@
+require_relative 'classes/item'
+
+# Class representing a Book
+class Book < Item
+  attr_accessor :publisher, :cover_state
+
+  def initialize(params = {})
+    super(params)
+    @publisher = params[:publisher]
+    @cover_state = params[:cover_state]
+  end
```

Comment:
Given the item class implementation below,  you need to adjust your constructor method to conform to the parent class (item), I realised using symbols for key will lead to issues when trying to rebuild the classes from json

```ruby
require 'securerandom'
require 'date'

class Item
  attr_accessor :genre, :author, :source, :label, :publish_date, :archived
  attr_reader :id

  def initialize(params = {})
    @id = params['id'] || SecureRandom.uuid
    @genre = params['genre'] || nil
    @author = params['author'] || nil
    @source = params['source'] || nil
    @label = params['label'] || nil
    @publish_date = params['publish_date']
    @archived = params['archived'] || false
  end

  def can_be_archived?
    publish_date = Date.parse(@publish_date)
    return false unless publish_date < (Date.today - (10 * 365))

    true
  end

  def move_to_archive
    @archived = true if can_be_archived?
  end
end
```